pkg/operator/ceph/object/config.go

```diff
@@ -61,28 +61,46 @@ func (c *clusterConfig) defaultSettings() *cephconfig.Config {
 		Set("rgw log nonexistent bucket", "true").
 		Set("rgw intent log object name utc", "true").
 		Set("rgw enable usage log", "true").
-		Set("rgw frontends", fmt.Sprintf("%s port=%s", rgwFrontend(c.clusterInfo.CephVersion), c.portString())).
+		Set("rgw frontends", fmt.Sprintf("%s %s", rgwFrontend(c.clusterInfo.CephVersion), c.portString(c.clusterInfo.CephVersion))).
 		Set("rgw zone", c.store.Name).
 		Set("rgw zonegroup", c.store.Name)
 	return s
 }
 
-func (c *clusterConfig) portString() string {
+func (c *clusterConfig) portString(v cephver.CephVersion) string {
 	var portString string
+
 	port := c.store.Spec.Gateway.Port
 	if port != 0 {
-		portString = strconv.Itoa(int(port))
+		portString = fmt.Sprintf("port=%s", strconv.Itoa(int(port)))
 	}
 	if c.store.Spec.Gateway.SecurePort != 0 && c.store.Spec.Gateway.SSLCertificateRef != "" {
-		var separator string
-		if port != 0 {
-			separator = "+"
-		}
 		certPath := path.Join(certDir, certFilename)
-		// with ssl enabled, the port number must end with the letter s.
-		// e.g., "443s ssl_certificate=/etc/ceph/private/keyandcert.pem"
-		portString = fmt.Sprintf("%s%s%ds ssl_certificate=%s",
-			portString, separator, c.store.Spec.Gateway.SecurePort, certPath)
+		// This is the beast backend
+		// Config is: http://docs.ceph.com/docs/master/radosgw/frontends/#id3
+		if v.IsAtLeastNautilus() {
```

Comment:
should this be be based on a beast feature flag rather than the version? i thought in nautilus civetweb was still optional?

Reply:
@noahdesu Your suggestion might be considered a new feature request. What would you think opening an issue for it? The current issue at hand prevents SSL-enabled RGW from working in 1.0.3. Also note that another bug prevented the use of beast in 1.0.0 thru 1.0.2. So it would be nice to get 1.0.4 released with this fix ASAP so that rook-ceph 1.0.x delivers a production-grade RGW beast frontend.

Reply:
@noahdesu this makes sense but Rook has decided to use Beast as of Nautilus.

Reply:
@leseb sounds good. the wording using "default" seemed to suggest it was optional https://github.com/rook/rook/pull/2708